src/instance/LeafInstance.ts

```diff
@@ -71,10 +80,14 @@ export default class LeafInstance implements Searchable, WithAttributes {
 
   public toJSON(authorized: Authorized, camelCase = false, convert = true): { [name: string]: LeafJSONValue } {
     return {
-      [this.model.getName(camelCase)]: convert ? this.getConvertedValue(authorized) : this.value
+      [this.model.getName(camelCase)]: convert ? this.getConvertedValue(authorized) : this.getValue(authorized)
     };
   }
 
+  public mapToJSON(authorized: Authorized, map: JSONMapper = x => x.toJSON(authorized)) {
```

Comment:
I think I'm a little confused by this api...

The top-level `mapToJson` takes two arguments, both functions. The first one returns a boolean whether the leaf can be converted. The second (which is optional,) is the actual map operation, _but_ the default value for it is to convert to JSON using the `authorized` argument?

But if I pass my own `map` then there's no way to access `authorized`? Which I guess is OK because the caller to `mapToJson` has access to `authorized` in their scope... but what if I don't care about `authorized`? Then I just pass a dummy function?

I think I would expect the signature to just be `mapToJSON(map: JSONMapper)`

Reply:
So yeah we export an `allow` function from `yinz` for that purpose.  The intent here is to make most of the access function not trust be default to prevent callers from accidentally leaking data. It does make the API more cumbersome, but the hope is that it makes it less prone to programmer error. We can then also grep through a codebase for `allow` to see where we explicitly let data pass through.